src/extract.js

```diff
@@ -81,20 +81,34 @@ function preprocessTemplate(data, type) {
   case 'jade':
   case 'pug':
     // Add empty require function to the context to avoid errors with webpack require inside pug
-    templateData = pug.render(data, {filename: 'source.html', pretty: true, require: function() {}});
+    templateData = pug.render(data, {filename: 'source.html', pretty: true, require: function() {}}).trim();
     break;
   case 'vue':
-    const vueFile = vueCompiler.parse({ compiler, source: data, needMap: false});
-    if (!vueFile.template) return '';  // return an empty string
-    templateData = vueFile.template.content;
-    if (vueFile.template.lang) {
-      return preprocessTemplate(templateData, vueFile.template.lang);
+    const $ = cheerio.load(templateData, {
+      xmlMode: true,
+      decodeEntities: false,
+      withStartIndices: true,
+    });
+
+    templateData = $('template').map(function() {
+      const t = $.html($(this));
+      const vueFile = vueCompiler.parse({ compiler, source: t, needMap: false});
+
+      if (vueFile.template.lang) {
+        return preprocessTemplate(($(this)).html(), vueFile.template.lang);
+      }
+      return vueFile.template.content.trim();
+    }).toArray();
+
+    // if there is just one template, use it as a string
+    if (templateData.length === 1) {
+      templateData = templateData[0];
```

Comment:
it's unnecessary to do this since you're modifying the result into an array anyway later [REF1]

Or did I miss something ?

Reply:
If I return an array here all the tests break. It'd need a major refactoring.

Reply:
interesting... OK !